src/Compilers/CSharp/Portable/Symbols/SubstitutedTypeParameterSymbol.cs

```diff
@@ -100,7 +100,7 @@ internal override ImmutableArray<TypeSymbolWithAnnotations> GetConstraintTypes(C
         {
             var constraintTypes = ArrayBuilder<TypeSymbolWithAnnotations>.GetInstance();
             _map.SubstituteTypesDistinctWithoutModifiers(_underlyingTypeParameter.GetConstraintTypes(inProgress, early), constraintTypes, null);
-            return constraintTypes.ToImmutableAndFree().WhereAsArray(type => type.SpecialType != SpecialType.System_Object || !type.IsAnnotated);
+            return constraintTypes.ToImmutableAndFree().WhereAsArray(type => type.SpecialType != SpecialType.System_Object || !type.NullableAnnotation.IsAnyNullable());
```

Comment:
>type.SpecialType != SpecialType.System_Object || !type.NullableAnnotation.IsAnyNullable() [](start = 77, length = 89)

Would it make sense to extract this logic out to a helper function (maybe `type => !type.IsNullableObject()`)? It also appears in `PETypeParameterSymbol.GetDeclaredConstraintTypes` as part of this PR.

Reply:
> Would it make sense to extract this logic out to a helper function (maybe type => !type.IsNullableObject())? It also appears in PETypeParameterSymbol.GetDeclaredConstraintTypes as part of this PR.

I personally do not see much value doing this at this point.

---
In reply to: [230539531](https://github.com/dotnet/roslyn/pull/30913#discussion_r230539531) [](ancestors = 230539531)